### ISSUE TITLE
Parsing error: '(' expected about  `baz instanceof Foo<Bar>`

### ISSUE THREAD

**Issue by yjhmelody** [eslint/eslint#14331]
**Tell us about your environment**

* **ESLint Version: v7.15.0**
* **Node Version: v15.3.0**
* **npm Version: 7.5.2**
* **Operating System: window 10** 

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

```js
            rules: {
                // Namespaces are quite useful in AssemblyScript
                "@typescript-eslint/no-namespace": "off",
  
                // There is actually codegen difference here
                "@typescript-eslint/no-array-constructor": "off",
  
                // Sometimes it can't be avoided to add a @ts-ignore
                "@typescript-eslint/ban-ts-comment": "off",
  
                // Utilized to achieve portability in some cases
                "@typescript-eslint/no-non-null-assertion": "off",

                "sort-imports": "warn",
                "no-multiple-empty-lines": "warn",
                "no-new-object": "error",
                "no-tabs": "warn",
                "no-trailing-spaces": "warn",
                "no-whitespace-before-property": "warn",
                "object-curly-newline": "warn",
                "semi": "warn",
                "sort-vars": "warn",
                "quotes": "warn",
                "space-in-parens": "warn",
                "space-infix-ops": "warn",
                "spaced-comment": ["warn", "always"],
                "space-before-function-paren": ["warn", "never"]
            }
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
baz instanceof Foo<Bar>
```

```bash
eslint . --fix
```

**What did you expect to happen?**
It could handle this ts syntax.

**What actually happened? Please copy-paste the actual, raw output from ESLint.**
```
122:0  error  Parsing error: '(' expected
```

**Comment by aladdin-add** [comment]
to support ts, you can install `@typescript-eslint/parser`, and config `parser: '@typescript-eslint/parser'`

**Comment by yjhmelody** [comment]
I used it.
```js
    parser: "@typescript-eslint/parser",
    plugins: [
        "@typescript-eslint",
    ],
    extends: [
        "eslint:recommended",
        "plugin:@typescript-eslint/eslint-recommended",
        "plugin:@typescript-eslint/recommended",
    ],
```

**Comment by aladdin-add** [comment]
I see. `baz instanceof Foo<Bar>` is not a valid ts, do you mean `baz instanceof Foo<Bar>()`?
